demos/defender/defender_demo_json/metrics_collector.c

```diff
@@ -378,4 +378,152 @@ MetricsCollectorStatus_t GetEstablishedConnections( Connection_t * pOutConnectio
+        status = MetricsCollectorBadParameter;
+    }
+
+    if( status == MetricsCollectorSuccess )
+    {
+        fileHandle = fopen( "/proc/uptime", "r" );
+
+        if( fileHandle == NULL )
+        {
+            LogError( ( "Failed to open /proc/uptime." ) );
+            status = MetricsCollectorFileOpenFailed;
+        }
+    }
+
+    if( status == MetricsCollectorSuccess )
+    {
+        if( fgets( &( lineBuffer[ 0 ] ), MAX_LINE_LENGTH, fileHandle ) != NULL )
+        {
+            float upTime = 0.0f, idleTime = 0.0f;
+            LogDebug( ( "File: /proc/uptime, Content: %s.",
+                        &( lineBuffer[ 0 ] ) ) );
+
+            /* Parse the output. */
+            filledVariables = sscanf( &( lineBuffer[ 0 ] ),
+                                      "%f %f",
+                                      &( upTime ),
+                                      &( idleTime ) );
+
+            /* sscanf should fill all the 2 variables successfully. */
+            if( filledVariables != 2 )
+            {
+                LogError( ( "Failed to CPU usage data. File: /proc/uptime, Data: %s.", &( lineBuffer[ 0 ] ) ) );
+                status = MetricsCollectorParsingFailed;
+            }
+            else
+            {
+                /* Convert data from floating point to integer by multiplying by 100 to represent data in
+                 * USER_HZ time units. */
+                pCpuUsage->upTime = ( int64_t ) ( upTime * 100.0f );
+                pCpuUsage->idleTime = ( int64_t ) ( idleTime * 100.0f );
+            }
+        }
+    }
+
+    if( fileHandle != NULL )
+    {
+        fclose( fileHandle );
+    }
+
+    return status;
+}
 /*-----------------------------------------------------------*/
+
+MetricsCollectorStatus_t GetMemoryStats( MemoryStats_t * pMemoryData )
+{
+    MetricsCollectorStatus_t status = MetricsCollectorSuccess;
+    FILE * fileHandle = NULL;
+    /* Variables for reading and processing data from "/proc/meminfo" file. */
+    uint8_t lineNumber = 0;
+    char lineBuffer[ MAX_LINE_LENGTH ];
+    bool readTotalMem = false, readAvailableMem = false;
+    const char * const pTotalMemoryField = "MemTotal";
+    const char * const pAvailableMemField = "MemAvailable";
+    int filledVariables = 0;
+    uint64_t parsedMemData = 0UL;
+
+    if( ( pMemoryData == NULL ) )
+    {
+        LogError( ( "Invalid parameter. pMemoryData=%p", ( void * ) pMemoryData ) );
+        status = MetricsCollectorBadParameter;
+    }
+
+    if( status == MetricsCollectorSuccess )
+    {
+        fileHandle = fopen( "/proc/meminfo", "r" );
+
+        if( fileHandle == NULL )
+        {
+            LogError( ( "Failed to open /proc/meminfo." ) );
+            status = MetricsCollectorFileOpenFailed;
+        }
+    }
+
+    if( status == MetricsCollectorSuccess )
+    {
+        while( ( fgets( &( lineBuffer[ 0 ] ), MAX_LINE_LENGTH, fileHandle ) != NULL ) )
+        {
+            lineNumber++;
```

Comment:
`lineNumber` is not needed.

Reply:
It is being printed in the debug and error log messages while lines are read from `proc/meminfo` just like other functions read multiple lines

Reply:
The only other function that has `lineNumber` is `GetEstablishedConnections` which uses it to skip the first line.